File: client/src/app/_models/params/userParams.ts
import { User } from "../user";
import { PaginationParams } from "./paginationParams";

export class UserParams extends PaginationParams {
    minAge: number = 18;
    maxAge: number = 60;
    gender: string;
    orderBy: string = 'lastActive';

    constructor(user: User | null) {
        super();
        this.gender = user?.gender === 'female' ? 'male' : 'female';
    }
}

File: API/Helpers/Params/UserParams.cs
﻿namespace API.Helpers.Params;

public class UserParams : PaginationParams
{
    public string? CurrentUsername { get; set; }

    public int MinAge { get; set; } = 18;

    public int MaxAge { get; set; } = 60;

    public string? Gender { get; set; }

    public string OrderBy { get; set; } = "lastActive";
}
